pkg/cloudprovider/providers/azure/auth/azure_auth.go

```diff
@@ -44,28 +44,39 @@ type AzureAuthConfig struct {
 	AADClientCertPassword string `json:"aadClientCertPassword" yaml:"aadClientCertPassword"`
 	// Use managed service identity for the virtual machine to access Azure ARM APIs
 	UseManagedIdentityExtension bool `json:"useManagedIdentityExtension" yaml:"useManagedIdentityExtension"`
+	// UserAssignedMSI contains the Client ID of the user assigned MSI which is assigned to the underlying VMs. If empty the user assigned identity is not used.
+	// More details of the user assigned identity can be found at: https://docs.microsoft.com/en-us/azure/active-directory/managed-service-identity/overview
+	// For the user assigned identity specified here to be used, the UseManagedIdentityExtension has to be set to true.
+	UserAssignedMSI string `json:"userAssignedMSI" yaml:"userAssignedMSI"`
```

Comment:
use UserAssignedIdentityID for clear? UserAssignedMSI seems overlapping with UseManagedIdentityExtension.